prawns/governance/src/handlers/treasury.proposed.event.ts

```diff
@@ -46,21 +51,25 @@ export default (network: SubstrateNetwork) =>
 
     // beneficiary and value
     try {
-      const call = getTreasuryProposedSpendCall(
-        <CallHandlerContext>ctx,
-        network
-      );
-      value = call.value;
-      beneficiary = decodeAddress(call.beneficiary);
+      const callHandler = createCallHandlerFromEventHandler(ctx);
+      if (!callHandler) {
+        throw new Error(
+          'Could not create callHandlerContext from EventHandlerContext'
+        );
+      }
+
+      const callArgs = getTreasuryProposedSpendCall(callHandler, network);
+      value = callArgs.value;
+      beneficiary = decodeAddress(callArgs.beneficiary);
       beneficiaryAccount = await getOrCreateGovernanceAccount(ctx.store, {
-        id: encodeAddress(network, call.beneficiary),
+        id: encodeAddress(network, callArgs.beneficiary),
         publicKey: beneficiary,
         network,
       });
       await ctx.store.save(beneficiaryAccount);
     } catch (e) {
```

Comment:
I think we should no longer need this catch

Reply:
We actually do because inside the `getTreasuryProposedSpendCall` we get an error in case the call wasn't of the type:
` assert(call.name === 'Treasury.propose_spend')`

```
treasury.proposed event: extrinsic hidden in wrapped call - Multisig.as_multi, not setting beneficiary or value fields
```

If so, we give up and don't go any further as back then we were afraid of multisig calls 😂

Reply:
What types can it be other than `Treasury.propose_spend`? I thought firesquid was handling all the proxy calls?

Reply:
Here's the nasty thing. We are looking at an event originally, not a call. 

The batch call or the multisig can generate multiple events, which will handle one by one. But if we're looking into the call inside the context, it's gonna be a batch one. So either we handle them, like I did here:
https://github.com/litentry/squid/pull/134/files#diff-ae8b19441920d35a5ed7230f9998cb72adcd680be4590d4b3e702541463e9b70R26

Or we ignore, which we are doing now :D